algorithms/ar-kmp/cpp/knuth_morris_pratt.cpp

```diff
@@ -0,0 +1,86 @@
+// C++ program for implementation of KMP pattern searching algorithm 
+#include <bits/stdc++.h> 
+
+// Fills lps[] for given patttern pat[0..M-1] 
+void LPSArray(char* pat, int M, int* lps) 
+{ 
+	
+	int len = 0; 
+
+	lps[0] = 0; 
+
+
+	int i = 1; 
+	while (i < M) { 
+		if (pat[i] == pat[len]) { 
+			len++; 
+			lps[i] = len; 
+			i++; 
+		} 
+		else // (pat[i] != pat[len]) 
+		{ 
+			 
+			if (len != 0) { 
+				len = lps[len - 1]; 
+
+				
+			} 
+			else // if (len == 0) 
+			{ 
+				lps[i] = 0; 
+				i++; 
+			} 
+		} 
+	} 
+} 
+
+// Prints occurrences of txt[] in pat[] 
+void KMP(char* pat, char* txt) 
+{ 
+	int M = strlen(pat); 
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6362540)